DuckDB.NET.Data/DuckDBSchema.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Linq;
+using System.Text;
+
+namespace DuckDB.NET.Data;
+
+internal static class DuckDBSchema
+{
+    public static DataTable GetSchema(DuckDBConnection connection, string collectionName, string?[]? restrictionValues) =>
+        collectionName.ToUpperInvariant() switch
+        {
+            "METADATACOLLECTIONS" => GetMetaDataCollections(),
+            "RESTRICTIONS" => GetRestrictions(),
+            "RESERVEDWORDS" => GetReservedWords(connection),
+            "TABLES" => GetTables(connection, restrictionValues),
+            "COLUMNS" => GetColumns(connection, restrictionValues),
+            "FOREIGNKEYS" => GetForeignKeys(connection, restrictionValues),
+            "INDEXES" => GetIndexes(connection, restrictionValues),
+            _ => throw new ArgumentOutOfRangeException(nameof(collectionName), collectionName, "Invalid collection name.")
+        };
+
+    private static DataTable GetMetaDataCollections() =>
+        new(DbMetaDataCollectionNames.MetaDataCollections)
+        {
+            Columns =
+            {
+                { DbMetaDataColumnNames.CollectionName, typeof(string) },
+                { DbMetaDataColumnNames.NumberOfRestrictions, typeof(int) },
+                { DbMetaDataColumnNames.NumberOfIdentifierParts, typeof(int) }
+            },
+            Rows =
+            {
+                { DbMetaDataCollectionNames.MetaDataCollections, 0, 0 },
+                { DbMetaDataCollectionNames.Restrictions, 0, 0 },
+                { DbMetaDataCollectionNames.ReservedWords, 0, 0 },
+                { DuckDbMetaDataCollectionNames.Tables, 4, 3 },
+                { DuckDbMetaDataCollectionNames.Columns, 4, 4 },
+                { DuckDbMetaDataCollectionNames.ForeignKeys, 4, 3 },
+                { DuckDbMetaDataCollectionNames.Indexes, 4, 3 },
```

Comment:
What do these numbers mean? Can you add a comment?

Reply:
This is from columns definition above:
```
                { DbMetaDataColumnNames.CollectionName, typeof(string) },
                { DbMetaDataColumnNames.NumberOfRestrictions, typeof(int) },
                { DbMetaDataColumnNames.NumberOfIdentifierParts, typeof(int) }
```

Reply:
What do the numbers 4 and 3 refer too?

Reply:
The first number says how many restrictions could the collection have (it is about `restrictionValues`/`restrictionNames` arrays). The last number is purely for information purposes. It says that index has 3 components in its full name: `catalog.schema.index_name`.